### ISSUE TITLE
Use my own data slice image for classification detection

### ISSUE THREAD

**Issue by LGC163** [hms-dbmi/CHIEF#46]
Hello, your work is very good, congratulations on your work on CHIEF, it is worth learning.

Yes, I have some private pathological section data images, and I want to test them with your CHIEF for classification and detection in the cancer cell detection section.

However, I have a small question:

- I understand that the code of that part is to directly detect the generated image slice pt file for prediction, what I want to know is how to convert my own data set image to the corresponding pt file, and what kind of pt file to convert, did you give it in your work?

Thank you.

**Comment by Dadatata-JZ** [comment]
Hi @LGC163 

Check this thread. 

https://github.com/hms-dbmi/CHIEF/issues/37#issuecomment-2528294290

**Comment by LGC163** [comment]
Thank you very much for taking time out of your busy schedule to reply me. @Dadatata-JZ 

So what you mean is that in the GET_CHIEF_patch_feature.py file, which I have read, there is indeed code about reading the data of.tif image type and output the tensor of the image. Here's my question:
1. Does that mean that our image data must be of.TIF type?
2. And how to save this as the corresponding pt file and save the name as the label in the corresponding csv data file?

I'd appreciate your advice.

```py
import torch, torchvision
import torch.nn as nn
from torchvision import transforms
from PIL import Image
from models.ctran import ctranspath

mean = (0.485, 0.456, 0.406)
std = (0.229, 0.224, 0.225)
trnsfrms_val = transforms.Compose(
    [
        transforms.Resize(224),
        transforms.ToTensor(),
        transforms.Normalize(mean = mean, std = std)
    ]
)


model = ctranspath()
model.head = nn.Identity()
td = torch.load(r'./model_weight/CHIEF_CTransPath.pth')
model.load_state_dict(td['model'], strict=True)
model.eval()


image = Image.open("./exsample/exsample.tif")
image = trnsfrms_val(image).unsqueeze(dim=0)
with torch.no_grad():
    patch_feature_emb = model(image) # Extracted features (torch.Tensor) with shape [1,768]
    print(patch_feature_emb.size())
```

**Comment by Dadatata-JZ** [comment]
Hi @LGC163 

(1) Nope. TIFF is just one of the supported formats—it’s not limited to CHIEF but applies to whole-slide images in general. The file types depend on what your scanning systems produce. Common formats include SVS, NDPI, and TIFF.

(2) You’ll need to run the standard pipeline, that is, tiling the WSIs and then extracting tile-level features using feature extractors like cTransPath, UNI, Phikon, Lunit, etc.

If you’re new to the field (computation pathology), it might help to check out some prior work that focuses on facilitating processing pipelines. Here is an example. There are tens more on git. 

https://github.com/PathologyDataScience/histolab?tab=readme-ov-file#getting-started

Lmk? Cheers

**Comment by guofeng97** [comment]
Hi! @Dadatata-JZ 
I have a problem running train_valid_test.py from Tumor_origin: 
fileNotFoundError: [Errno 2] No such file or directory: '. /feature/tcga/TCGA-US-A774-01Z-00-DX1.522FF138-153F-488A-BBBC-5EA68EFD80C7.pt'

In '. /feature/tcga' directory there are no corresponding .pt files, are these .pt files generated after you performed patch level feature extraction in Get_CHIEF_patch_feature.py? Do I need to download the dataset first and then run the images through feature extraction to generate the .pt files?

Looking forward to your reply!
